src/core/finder.ml

```diff
@@ -0,0 +1,63 @@
+(* 
+ * Module in charge of return 
+ * all files targeted by eos
+ * @author MARAIS - DURAND © 2019
+ *)
+
+(* This function checks if a string passed in argument is a directory. 
+    The default value is Sys.is_directory function.
+*)
+let check_directory path : bool = Sys.is_directory path
+
+(*
+   This function returns if a file is targeted.
+*)
+let is_targeted target path : bool =
+  let exist (el : string) : bool =
+    let el_len = String.length el in
+    let dir = if el_len > 0 then String.get el (String.length el - 1) = '/'  else false in
+    if dir && not (check_directory path) then false
+    else begin
+      let el = if dir then String.sub el 0 (String.length el - 1) else el in 
+      let regex = Str.regexp el in
+      try Str.search_backward regex path (String.length path - 1) > -1
+      with Not_found -> false
+    end
+  in
+  List.exists exist target
+
+(*
+   Tail-recursive search (and add to [l]) of all files in directories and sub-directories given at second argument. 
+   If the second argument is not a directory, we add it in the list [l].
+*)
+let rec get_sub_files l path : string list =
+  if not (check_directory path) then
```

Comment:
Why not merging this function and the one below ? Get_subdir should test if it's a targeted file...

Reply:
`get_sub_files` takes all subfiles without taking into account the list of regex. Whereas `get_files` takes it into account.
`get_sub_files` is called when a directory is matching (and we take all subfiles).

Reply:
There is a problem if we put `*.c` . I know it's annoying to change it now but I wonder if .eosignore is not a better way...